### ISSUE TITLE
AER model load after saving causes error

### ISSUE THREAD

**Issue by yaseryacoob** [sintel-dev/Orion#302]
@sarahmish 

I saved the AER model in two ways (1) using pickle (2) and using orion.save() and then orion.load().

In both cases I am getting an error on attempting to predict, see below. This error doesn't occur if I use the model immediately after fitting.  Looking at the code https://github.com/sintel-dev/Orion/blob/master/orion/primitives/aer.py I am not clear what the cause may be.  

Any suggestions?

```
  <orion.core.Orion object at 0x7f85e74e5828>
Exception caught producing MLBlock orion.primitives.aer.AER#1
Traceback (most recent call last):
  File "/fs/nexus-scratch/yaser/miniconda3/envs/orionGPU2/lib/python3.6/site-packages/mlblocks/mlpipeline.py", line 565, in _produce_block
    block_outputs = block.produce(**produce_args)
  File "/fs/nexus-scratch/yaser/miniconda3/envs/orionGPU2/lib/python3.6/site-packages/mlblocks/mlblock.py", line 322, in produce
    return getattr(self.instance, self.produce_method)(**produce_kwargs)
  File "/fs/nexus-scratch/yaser/time-series/Orion/orion/primitives/aer.py", line 196, in predict
    ry, y, fy = self.aer_model.predict(X)
AttributeError: 'AER' object has no attribute 'aer_model'
Traceback (most recent call last):
  File "yy.py", line 94, in <module>
    anomalies = orion.detect(train_data)
  File "/fs/nexus-scratch/yaser/time-series/Orion/orion/core.py", line 150, in detect
    return self._detect(self._mlpipeline.predict, data, visualization)
  File "/fs/nexus-scratch/yaser/time-series/Orion/orion/core.py", line 113, in _detect
    outputs = method(data, output_=outputs_spec, **kwargs)
  File "/fs/nexus-scratch/yaser/miniconda3/envs/orionGPU2/lib/python3.6/site-packages/mlblocks/mlpipeline.py", line 722, in predict
    self._produce_block(block, block_name, context, output_variables, outputs)
  File "/fs/nexus-scratch/yaser/miniconda3/envs/orionGPU2/lib/python3.6/site-packages/mlblocks/mlpipeline.py", line 565, in _produce_block
    block_outputs = block.produce(**produce_args)
  File "/fs/nexus-scratch/yaser/miniconda3/envs/orionGPU2/lib/python3.6/site-packages/mlblocks/mlblock.py", line 322, in produce
    return getattr(self.instance, self.produce_method)(**produce_kwargs)
  File "/fs/nexus-scratch/yaser/time-series/Orion/orion/primitives/aer.py", line 196, in predict
    ry, y, fy = self.aer_model.predict(X)
AttributeError: 'AER' object has no attribute 'aer_model'



```

**Comment by sarahmish** [comment]
Thank you @yaseryacoob for highlighting this issue!

Fixed in PR #304.

**Comment by yaseryacoob** [comment]
Unfortunately it is not working, I get a new error on loading the model, I pulled from github all changes, here is my code and the error.

```
train_data = load_signal('Honorable_Interview_00_02_04_EAR_R')

hyperparameters = {
    'mlprimitives.custom.timeseries_preprocessing.time_segments_aggregate#1': {
        'interval': 1
    },
    'orion.primitives.aer.AER#1': {
        'epochs': 3,
        'verbose': True ,
        'input_shape': [200, 1] #,
    }
  }


orion = Orion(
    pipeline='aer',
    hyperparameters=hyperparameters
)

if 1:
  orion.fit(train_data)
  orion.save("AER/temp00AER.pkl")
  with open("tempAER.pkl", "wb") as pickle_file:
      pickle.dump(orion, pickle_file)
else:
   orion = pd.read_pickle("tempAER.pkl")
   orion = Orion( pipeline='aer')
   orion=orion.load("AER/temp00AER.pkl")
```



```
(orionGPU3) yaser@yaserubuntu:/media/ydrive/time_series/Orion0.3.1$ !py
python yydebugAER.py
/home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/sklearn/utils/validation.py:37: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  LARGE_SPARSE_SUPPORTED = LooseVersion(scipy_version) >= '0.14.0'
Using TensorFlow backend.
WARNING:tensorflow:From /home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/tensorflow/python/ops/init_ops.py:97: calling GlorotUniform.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.
Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor
WARNING:tensorflow:From /home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/tensorflow/python/ops/init_ops.py:1251: calling VarianceScaling.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.
Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor
WARNING:tensorflow:From /home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/tensorflow/python/ops/init_ops.py:97: calling Orthogonal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.
Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor
WARNING:tensorflow:From /home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/tensorflow/python/ops/init_ops.py:97: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.
Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor
Traceback (most recent call last):
  File "/home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/pandas/io/pickle.py", line 166, in try_read
    return read_wrapper(lambda f: pkl.load(f))
  File "/home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/pandas/io/pickle.py", line 148, in read_wrapper
    return func(f)
  File "/home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/pandas/io/pickle.py", line 166, in <lambda>
    return read_wrapper(lambda f: pkl.load(f))
  File "/media/ydrive/time_series/Orion0.3.1/orion/primitives/aer.py", line 76, in __setstate__
    state[network] = tf.keras.models.load_model(fd.name)
  File "/home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/tensorflow/python/keras/saving/save.py", line 146, in load_model
    return hdf5_format.load_model_from_hdf5(filepath, custom_objects, compile)
  File "/home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/tensorflow/python/keras/saving/hdf5_format.py", line 212, in load_model_from_hdf5
    custom_objects=custom_objects)
  File "/home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/tensorflow/python/keras/saving/model_config.py", line 55, in model_from_config
    return deserialize(config, custom_objects=custom_objects)
  File "/home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/tensorflow/python/keras/layers/serialization.py", line 89, in deserialize
    printable_module_name='layer')
  File "/home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/tensorflow/python/keras/utils/generic_utils.py", line 192, in deserialize_keras_object
    list(custom_objects.items())))
  File "/home/yaser/miniconda3/envs/orionGPU3/lib/python3.7/site-packages/tensorflow/python/keras/engine/network.py", line 1152, in from_config
    layer_output_tensors = layer._inbound_nodes[node_index].output_tensors
IndexError: list index out of range

During handling of the above exception, another exception occurred:

```
